llpc/context/llpcCompiler.cpp

```diff
@@ -1418,12 +1429,162 @@ Result Compiler::buildGraphicsPipelineInternal(GraphicsContext *graphicsContext,
   }
 
   if (result != Result::Success) {
-    result = buildPipelineInternal(context, shaderInfo, /*unlinked=*/false, pipelineElf, stageCacheAccesses);
+    // See if we want to use part-pipeline compilation, if there is an FS and at least one pre-rasterization
+    // shader stage.
+    unsigned stageMask = context->getShaderStageMask();
+    bool buildPartPipeline = (cl::EnablePartPipeline && (stageMask & (1 << ShaderStageFragment)) &&
+                              (stageMask & ~(1 << ShaderStageFragment)));
```

Comment:
It would be much better to use helper function to convert enums to/from masks, check if a stage is contained in a mask or not. This is not only more readable but can also be unit-tested for unobvious edge cases.

This also applies to similar code patterns in the rest of the PR.

Reply:
Converted to `shaderStageToMask()`.